src/main/java/org/radarcns/management/web/rest/SubjectResource.java

```diff
@@ -101,9 +119,25 @@
      */
     @GetMapping("/subjects")
     @Timed
-    public List<SubjectDTO> getAllSubjects() {
-       log.debug("REST request to get all Subjects");
-       return subjectService.findAll();
+    public ResponseEntity getAllSubjects(
+        @RequestParam(value = "projectId" , required = false) Long projectId,
+        @RequestParam(value = "externalId" , required = false) String externalId) {
+        log.error("ProjectID {} and external {}" , projectId, externalId);
+        if(projectId!=null && externalId!=null) {
+            Subject subject = subjectRepository.findOneByProjectIdAndExternalId(projectId, externalId).get();
+            SubjectDTO subjectDTO = subjectMapper.subjectToSubjectDTO(subject);
+            return ResponseUtil.wrapOrNotFound(Optional.ofNullable(subjectDTO));
```

Comment:
I think we should return a `List<SubjectDTO>` here as well. This way consumers of the API know what the response type is and know it will always be the same. Otherwise a client would always need to test for list or single entity. We can create a list from a single element easily with e.g. `Collections.singletonList(subjectDTO)`. Then you can also make the type signature more specific to `ResponseEntity<List<SubjectDTO>>`.